projects/scannet_offline_eval/demo/proto_robot/robodata_pb2_grpc.py

```diff
@@ -0,0 +1,213 @@
+        context.set_details("Method not implemented!")
+        raise NotImplementedError("Method not implemented!")
+
+    def Chat(self, request, context):
+        """Missing associated documentation comment in .proto file."""
+        context.set_code(grpc.StatusCode.UNIMPLEMENTED)
+        context.set_details("Method not implemented!")
+        raise NotImplementedError("Method not implemented!")
+
+
+def add_RobotDataServicer_to_server(servicer, server):
+    rpc_method_handlers = {
+        "GetHistory": grpc.unary_stream_rpc_method_handler(
+            servicer.GetHistory,
+            request_deserializer=robodata__pb2.RoboTensor.FromString,
+            response_serializer=robodata__pb2.RobotSummary.SerializeToString,
+        ),
+        "ReceiveRobotData": grpc.stream_stream_rpc_method_handler(
+            servicer.ReceiveRobotData,
+            request_deserializer=robodata__pb2.RobotSummary.FromString,
+            response_serializer=robodata__pb2.RobotSummary.SerializeToString,
+        ),
+        "PlanHighLevelAction": grpc.stream_stream_rpc_method_handler(
+            servicer.PlanHighLevelAction,
+            request_deserializer=robodata__pb2.RobotSummary.FromString,
+            response_serializer=robodata__pb2.RobotSummary.SerializeToString,
+        ),
+        "Chat": grpc.unary_stream_rpc_method_handler(
+            servicer.Chat,
+            request_deserializer=robodata__pb2.LLMInput.FromString,
+            response_serializer=robodata__pb2.ChatMsg.SerializeToString,
+        ),
+    }
+    generic_handler = grpc.method_handlers_generic_handler(
+        "robodata.RobotData", rpc_method_handlers
+    )
+    server.add_generic_rpc_handlers((generic_handler,))
+
+
+# This class is part of an EXPERIMENTAL API.
+class RobotData(object):
+    """Interface exported by the server."""
+
+    @staticmethod
+    def GetHistory(
+        request,
+        target,
+        options=(),
+        channel_credentials=None,
+        call_credentials=None,
+        insecure=False,
+        compression=None,
+        wait_for_ready=None,
+        timeout=None,
+        metadata=None,
+    ):
+        return grpc.experimental.unary_stream(
+            request,
+            target,
+            "/robodata.RobotData/GetHistory",
+            robodata__pb2.RoboTensor.SerializeToString,
+            robodata__pb2.RobotSummary.FromString,
+            options,
+            channel_credentials,
+            insecure,
+            call_credentials,
+            compression,
+            wait_for_ready,
+            timeout,
+            metadata,
+        )
+
+    @staticmethod
+    def ReceiveRobotData(
+        request_iterator,
+        target,
+        options=(),
+        channel_credentials=None,
+        call_credentials=None,
+        insecure=False,
+        compression=None,
+        wait_for_ready=None,
+        timeout=None,
+        metadata=None,
+    ):
+        return grpc.experimental.stream_stream(
```

Comment:
Not blocking -- why is this experimental?

Reply:
don't know, auto-generated

Reply:
I believe this is autogenerated code